osxkeychain/osxkeychain_darwin.go

```diff
@@ -168,3 +164,25 @@ func freeServer(s *C.struct_Server) {
 	C.free(unsafe.Pointer(s.host))
 	C.free(unsafe.Pointer(s.path))
 }
+
+// parseURL parses and validates a given serverURL to an url.URL, and
+// returns an error if validation failed. Querystring parameters are
+// omitted in the resulting URL, because they are not used in the helper
+func parseURL(serverURL string) (*url.URL, error) {
+	if !regexp.MustCompile(`^([a-zA-Z][-+.a-zA-Z0-9]+:)?//`).MatchString(serverURL) {
+		serverURL = "//" + serverURL
+	}
+	u, err := url.Parse(serverURL)
+	if err != nil {
+		return nil, err
+	}
+	if u.Scheme != "" && u.Scheme != "https" && u.Scheme != "http" {
+		return nil, errors.New("unsupported scheme: " + u.Scheme)
+	}
+	if u.Hostname() == "" {
+		return nil, errors.New("no hostname in URL")
+	}
+
+	u.RawQuery = ""
```

Comment:
Explicitly strip query-string, because we're not storing it, but let me know if you don't want this 👍